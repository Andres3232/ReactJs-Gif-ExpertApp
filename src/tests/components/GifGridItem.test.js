

import { shallow } from 'enzyme';
import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas con GifGridItem()', () => {

    const title = 'Titulo'
    const url = 'https://urlmock'
    const wrapper = shallow( <GifGridItem title={title} url={url}/>)
    
    test('debe mostrar <GifGridItem/> correctamente', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('debe de tener un parrafo con el tittle', () => {
        const p =wrapper.find('p');
        expect ( p.text().trim() ).toBe( title )
    })

    test('debe de tener la imagen igual al url y alt de las props', () => {
        
        const img = wrapper.find('img')
        //console.log( img.prop('src') );
        expect( img.prop('src') ).toBe( url )
        expect( img.prop('alt') ).toBe( title )
    })

    test('debe de tener animate__fadeIn', () => {
        
        const div = wrapper.find('div')
        const className = div.prop('className');
    
        expect(className.includes('animate__fadeIn')).toBe(true)


        
    })
    
    
    
    
})
