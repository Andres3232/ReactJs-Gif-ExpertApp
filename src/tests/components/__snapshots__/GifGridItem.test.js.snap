// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con GifGridItem() debe mostrar <GifGridItem/> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://urlmock"
  />
  <p>
     
    Titulo
  </p>
</div>
`;
